pages/profile/index.tsx

```diff
@@ -53,35 +76,74 @@ const ProfilePage: FC = () => {
     });
   };
 
-  const handleSubmit = useCallback(async (evt) => {
+  const handleSave = useCallback(async (evt) => {
     evt.preventDefault();
 
     const {
-      username,
+      username: name,
     } = credentials;
 
     const {
       token,
     } = user;
 
     try {
-      await updateUser({ username }, token);
+      await updateUser({ username: name }, token);
       queryClient.invalidateQueries('me');
     } catch (e) {
       throw new Error(`something went wrong updating user: ${e.message}`);
     }
   }, [credentials, user, queryClient]);
 
+  const handlePasswordChange = useCallback(async (evt) => {
+    evt.preventDefault();
+
+    const {
+      token,
+    } = user;
+
+    try {
+      if (newPassword !== confirmation) {
+        setPasswordsMismatchMessage(true);
+        setTimeout(() => setPasswordsMismatchMessage(false), 3000);
+      }
+      await updateUser({ username }, token);
```

Comment:
You could use a [mutation](https://react-query.tanstack.com/reference/useMutation) here.